test/b7-shared-data-component.js

```diff
@@ -107,6 +107,22 @@ describe('b7 - shared data component', function () {
       });
     });
 
+    it('respects noPublish set option', function (done) {
+      meshInstance.event.data.on('/some/path/testNoPublish', function (data, meta) {
+        delete data._meta;
+        data.should.eql({val: 'must be emitted'});
+        done();
+      }, function (e) {
+        if (e) return done(e);
+        dataComponent.set('/some/path/testNoPublish', {val: 'must not be emitted'}, {noPublish: true}, function (e) {
```

Comment:
How do you know this one didn't fire?

Reply:
there would still be an exception - but I can improve on this

Reply:
made the test wait for a second before exiting; if I force the incorrect value to be emitted by setting noPublish to false, the test fails as expected (which it also did before)